src/stores/MessageList.tsx

```diff
@@ -0,0 +1,160 @@
+			};
+		}
+	};
+
+	const jsonMessagesFromString = (rawFromFile: string | ArrayBuffer | null) => {
+		const json = JSON.parse(rawFromFile as string);
+		clearParsedMessages();
+
+		for (let i = 0; i < json.length; i++) {
+			addParsedMessage(JSON.stringify(json[i]));
+		}
+	};
+
+	const replay = (array: string[]) => {
+		setFromFile(true);
+		clearParsedMessages();
+		let longestDelay = 0;
+		array.forEach((item, index) => {
+			if (JSON.parse(item).delay && JSON.parse(item).delay >= longestDelay) {
+				longestDelay = index;
+			}
+		});
+
+		for (let i = 0; i < array.length; i++) {
+			if (JSON.parse(array[i]).delay) {
+				// eslint-disable-next-line no-loop-func
+				setTimeout(() => {
+					sendMessage(array[i] as unknown as object);
+					if (i === longestDelay) {
+						setFromFile(false);
+					}
+				},
+				JSON.parse(array[i]).delay);
+			} else {
+				sendMessage(array[i] as unknown as object);
+			}
+		}
+	};
+
+	return (
+		<div>
+			<div>
+			</div>
+			<div style={{ height: '430px', overflowY: 'scroll' }}>
+			 <MessageList items={parsedMessagesHistory} indicators={indicators} editMessageMode={editMessageMode}
+					editedMessageIndex={editedMessageIndex}/>
+			</div>
+			<div>
+				<button className="mainButton" onClick={() => { clearParsedMessages(); }}>
+                        Clear
+				</button>
+				<button className="mainButton" >
+					<a
+						href={`data:application/json;base64,${btoa(`[${parsedMessagesHistory.toString()}]`)}`} download>
+                        Export</a>
+				</button>
+				<button className="mainButton" onClick={() => { replay(parsedMessagesHistory); }}>
+                    Replay
+				</button>
+				<input value={loadedFile} title=' ' id='file' type="file" accept='.json' onChange={ e => {
+					if (e !== null) {
+						fileLoader(e?.target.files);
+						setFile('');
+					}
+				} }/>
+			</div>
+		</div>
+	);
+};
+
+export const Indicators = (props: {indicator: string}) => (
+	<div>
+		<button className={props.indicator}>{props.indicator}</button>
+	</div>
+);
+
+export const MessageList = (props: {indicators: string[]; items: string[]; editMessageMode: boolean;
+    editedMessageIndex: number;}) => {
+	const {
+		parsedMessagesHistory, clearParsedMessages, deleteIndicator, addParsedMessage,
+		setEditMessageMode, setEditedMessageSendDelay, selectMessage, editMessageMode,
+		indicators,
+	} = useStore();
+
+	const deleteMessage = (index: number) => {
+		const newArray: string[] = [];
+		const tmpIndicators: string[] = deleteIndicator(index);
+		parsedMessagesHistory.forEach((item, i) => {
+			if (i !== index) {
+				newArray.push(item);
+			}
+		});
+		clearParsedMessages();
+		newArray.forEach((mess, i) => {
+			addParsedMessage(mess, tmpIndicators[i]);
+		});
+	};
+
+	const setDelay = (delay: number) => {
+		setEditedMessageSendDelay(delay);
+	};
+
+	return (
+		<ul>
+			<li key={props.items.length + 1}>
+				<div className={editMessageMode ? 'normalNewMessage' : 'hiddenNewMessage'}
```

Comment:
instead of chaning className to hide element it better to use logical operators
`{condition && <element/>}`